python/pylibcudf/pylibcudf/tests/test_column_from_device.py

```diff
@@ -69,6 +74,21 @@ def iface_obj(input_column):
 
 
 def test_from_cuda_array_interface(input_column, iface_obj):
-    col = plc.column.Column.from_cuda_array_interface_obj(iface_obj)
+    expect = input_column
+
+    res = plc.Column.from_cuda_array_interface_obj(iface_obj)
+
+    assert_column_eq(expect, res)
+
+
+def test_from_1d_cupy_array(request, valid_type, input_column):
+    request.applymarker(
+        pytest.mark.xfail(
+            condition=valid_type in DATETIME_TYPES,
+            reason="Datetime types are not yet supported in cupy. See cupy/issues/2622.",
+        )
+    )
+    expect = input_column
+    res = plc.Column.from_cuda_array_interface_obj(cp.asarray(expect))
```

Comment:
Could we just modify the `DataBuffer` object to have a `self.strides = None` so we can avoid this extra dependency?

Reply:
Yes, done in https://github.com/rapidsai/cudf/pull/18295/commits/b3bee38ea76b2f1db8761599675b0a637e2cb5b8